src/main/java/com/minecrafttas/mctcommon/Configuration.java

```diff
@@ -5,40 +5,75 @@
 import java.io.FileNotFoundException;
 import java.io.FileOutputStream;
 import java.io.IOException;
-import java.util.HashMap;
+import java.util.ArrayList;
 import java.util.InvalidPropertiesFormatException;
-import java.util.Map;
+import java.util.LinkedHashMap;
+import java.util.List;
 import java.util.Properties;
 
+import com.minecrafttas.mctcommon.registry.AbstractRegistry;
+import com.minecrafttas.mctcommon.registry.Registerable;
 
 /**
  * A <i>very</i> simple configuration class
+ * 
  * @author Scribble
- *
  */
 
-public class Configuration {
-	
+public class Configuration extends AbstractRegistry<com.minecrafttas.mctcommon.Configuration.ConfigOptions> {
+
```

Comment:
Why the long import string?

Reply:
The compiler threw an error and eclipse just refused to import it... Kinda forgot about it